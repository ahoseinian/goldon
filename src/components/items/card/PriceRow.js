import React from 'react'
import Disc from './Disc'

const PriceRow = ({price}) => price ? (
  <div className="pt-2 pl-3">
    <Disc name="قیمت" value={price.toLocaleString()} measure="تومان"/>
  </div>
): null

PriceRow.propTypes = {
  price: React.PropTypes.number
}

export default PriceRow
