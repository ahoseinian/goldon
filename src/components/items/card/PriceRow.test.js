import React from 'react'
import {shallow} from 'enzyme'
import PriceRow from './PriceRow'
import toJson from 'enzyme-to-json'

describe('PriceRow', () => {
  it('renders null without price', () => {
    const wrapper = shallow(<PriceRow />)
    expect(wrapper.html()).toEqual(null);
  })

  it('renders with price', () => {
    const wrapper = shallow(<PriceRow price={123}/>)
    expect(wrapper.html()).toContain('123');
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
