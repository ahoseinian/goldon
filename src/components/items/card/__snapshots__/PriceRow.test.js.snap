exports[`PriceRow renders with price 1`] = `
<div
  className="pt-2 pl-3">
  <Disc
    measure="تومان"
    name="قیمت"
    value="123" />
</div>
`;
